pkg/system/phase4_configuring.go

```diff
@@ -956,31 +956,34 @@ func (r *Reconciler) prepareGCPBackingStore() error {
 		return err
 	}
 	projectID := authJSON.ProjectID
-	r.GCPBucketCreds.StringData["GoogleServiceAccountPrivateKeyJson"] = cloudCredsSecret.StringData["service_account.json"]
-	ctx := context.Background()
-	gcpclient, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(cloudCredsSecret.StringData["service_account.json"])))
-	if err != nil {
-		r.Logger.Info(err)
-		return err
-	}
+	if r.GCPBucketCreds.StringData != nil {
+		r.GCPBucketCreds.StringData["GoogleServiceAccountPrivateKeyJson"] = cloudCredsSecret.StringData["service_account.json"]
 
-	var bucketName = strings.ToLower(randname.GenerateWithPrefix("noobaabucket", 5))
-	if err := r.createGCPBucketForBackingStore(gcpclient, projectID, bucketName); err != nil {
-		r.Logger.Info(err)
-		return err
-	}
+		ctx := context.Background()
+		gcpclient, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(cloudCredsSecret.StringData["service_account.json"])))
+		if err != nil {
+			r.Logger.Info(err)
+			return err
+		}
 
-	if errUpdate := r.Client.Update(r.Ctx, r.GCPBucketCreds); errUpdate != nil {
-		return fmt.Errorf("got error on GCPBucketCreds update. error: %v", errUpdate)
-	}
-	// create backing store
-	r.DefaultBackingStore.Spec.Type = nbv1.StoreTypeGoogleCloudStorage
-	r.DefaultBackingStore.Spec.GoogleCloudStorage = &nbv1.GoogleCloudStorageSpec{
-		TargetBucket: bucketName,
-		Secret: corev1.SecretReference{
-			Name:      r.GCPBucketCreds.Name,
-			Namespace: r.GCPBucketCreds.Namespace,
-		},
+		var bucketName = strings.ToLower(randname.GenerateWithPrefix("noobaabucket", 5))
+		if err := r.createGCPBucketForBackingStore(gcpclient, projectID, bucketName); err != nil {
+			r.Logger.Info(err)
+			return err
+		}
+
+		if errUpdate := r.Client.Update(r.Ctx, r.GCPBucketCreds); errUpdate != nil {
+			return fmt.Errorf("got error on GCPBucketCreds update. error: %v", errUpdate)
+		}
+		// create backing store
+		r.DefaultBackingStore.Spec.Type = nbv1.StoreTypeGoogleCloudStorage
+		r.DefaultBackingStore.Spec.GoogleCloudStorage = &nbv1.GoogleCloudStorageSpec{
+			TargetBucket: bucketName,
+			Secret: corev1.SecretReference{
+				Name:      r.GCPBucketCreds.Name,
+				Namespace: r.GCPBucketCreds.Namespace,
+			},
+		}
```

Comment:
Thanks, I changed the condition added a log message, and returned an error.
It would look like this in the logs:
```
time="2023-12-04T11:22:03Z" level=info msg="Secret \"noobaa-gcp-cloud-creds-secret\" does not contain a map of StringData yet. retry on next reconcile..." sys=test1/noobaa
time="2023-12-04T11:22:03Z" level=info msg="SetPhase: temporary error during phase \"Configuring\"" sys=test1/noobaa
time="2023-12-04T11:22:03Z" level=warning msg="⏳ Temporary Error: cloud credentials secret \"noobaa-gcp-cloud-creds-secret\" is not ready yet (does not contain a map of StringData yet)" sys=test1/noobaa
```